src/Extension/AccountReset/SecurityExtension.php

```diff
@@ -0,0 +1,204 @@
+use SilverStripe\Core\Extension;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormAction;
+use SilverStripe\Forms\PasswordField;
+use SilverStripe\Forms\RequiredFields;
+use SilverStripe\MFA\JSONResponse;
+use SilverStripe\MFA\RequestHandler\BaseHandlerTrait;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+
+/**
+ * Extends the Security controller to support Account Resets. This extension can
+ * itself be extended to add procedures to the reset action (such as removing
+ * additional authentication factors, sending alerts, etc.)
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property Security owner
+ */
+class SecurityExtension extends Extension
+{
+    use BaseHandlerTrait;
+    use Extensible;
+
+    private static $url_handlers = [
+        'GET reset-account' => 'resetaccount',
+    ];
+
+    private static $allowed_actions = [
+        'resetaccount',
+        'ResetAccountForm',
+    ];
+
+    public function resetaccount(HTTPRequest $request)
+    {
+        if (Security::getCurrentUser()) {
+            $output = $this->owner->renderWith(
+                'Security',
+                [
+                    'Title' => _t(
+                        __CLASS__ . '.ALREADYAUTHENTICATEDTITLE',
+                        'Already authenticated'
+                    ),
+                    'Content' => _t(
+                        __CLASS__ . '.ALREADYAUTHENTICATEDBODY',
+                        'You must be logged out to reset your account.'
+                    ),
+                ]
+            );
+            return $this->owner->getResponse()->setBody($output)->setStatusCode(400);
+        }
+
+        $vars = $request->getVars();
+
+        /** @var Member|MemberExtension $member */
+        $member = Member::get()->byID(intval($vars['m'] ?? 0));
+
+        if (is_null($member) || $member->verifyAccountResetToken($vars['t'] ?? '') === false) {
+            $output = $this->owner->renderWith(
+                'Security',
+                [
+                    'Title' => _t(
+                        __CLASS__ . '.INVALIDTOKENTITLE',
+                        'Invalid member or token'
+                    ),
+                    'Content' => _t(
+                        __CLASS__ . '.INVALIDTOKENBODY',
+                        'Your account reset token may have expired. Please contact an administrator.'
+                    )
+                ]
+            );
+            return $this->owner->getResponse()->setBody($output)->setStatusCode(400);
+        }
+
+        $request->getSession()->set('MemberID', $member->ID);
+
+        return $this->owner->getResponse()->setBody($this->owner->renderWith(
+            'Security',
+            [
+                'Title' => _t(
+                    __CLASS__ . '.ACCOUNT_RESET_TITLE',
+                    'Reset Account'
```

Comment:
I'd suggest a minor case change:

```suggestion
                    'Reset account'
```